build_runner_core/lib/src/asset_graph/graph.dart

```diff
@@ -277,7 +276,7 @@ class AssetGraph {
       }
     });
 
-    var newAndModifiedNodes = modifyIds.map(get).toList()
+    var newAndModifiedNodes = modifyIds.map((id) => get(id)!).toList()
```

Comment:
I think we could also do this with `.cast<AssetNode>()`?

It might look nicer though with a collection literal?

```suggestion
    var newAndModifiedNodes = [
        for (var id in modifyIds) get(id)!,
        ..._addSources(newIds),
    ];
```